### PR TITLE
Add additional server side handling events to testem config

### PR DESCRIPTION
**Pull request by choheekim**
Add additional server side handling events to testem config

### CODE REVIEW

File: lib/commands/exam.js [hunk 1]
```diff
@@ -163,6 +165,7 @@ module.exports = TestCommand.extend({
       config.testPage = this._generateTestPages(customBaseUrl, appendingParam, ids);
     }
 
+
```
Comment (step2yeung) [inline lib/commands/exam.js]: remove blank line?

File: lib/commands/exam.js [hunk 2]
```diff
@@ -313,5 +316,35 @@ module.exports = TestCommand.extend({
...
+   * @param {object} config
+   * @return {object} events
+   */
+  _addBrowserSocketEventsToConfig: function(config) {
+    let events = config.additionalBrowserSocketEvents || {};
+
+    events['set-modules-queue'] = function (modules) {
+      const proto = Object.getPrototypeOf(this)
+      if (!proto.moduleQueue) {
+        proto.moduleQueue = modules;
```
Comment (step2yeung) [inline lib/commands/exam.js]: Would be good to comment why we are using the prototype to save this list

File: lib/commands/exam.js [hunk 3]
```diff
@@ -313,5 +316,35 @@ module.exports = TestCommand.extend({
...
+      }
+    }
+
+    events['get-next-module'] = function () {
+      const proto = Object.getPrototypeOf(this)
+      const moduleName = proto.moduleQueue.shift();
```
Comment (step2yeung) [inline lib/commands/exam.js:347]: You can just use `moduleQueue.shift();` here

Comment (choheekim) [reply lib/commands/exam.js:347]: hm wouldn't moduleQueue be undefined?

Comment (step2yeung) [reply lib/commands/exam.js:347]: good call, you may need to use `this.moduleQueue.shift()`

**Comment by scalvert** [comment]
Can we open these against `ember-cli/ember-exam` instead of this repo. It's going to be more work/time to PR against this repo, then have to PR again against upstream. Additionally, we want to get eyes on this code as early as possible. 

I'd suggest we be very explicit, in the PR description for _all_ the PRs that are related, what the sequence is for landing these changes.

For example - possible PR titles and description:

Title: [Part 1] - Load balancing Ember Exam - add weighted option
Description: Adds weighted option to ember exam to provide better utilization of browser parallelization.

This change is Part 1 in a 5 part series:
- Part 1 - add weighted option
- Part 2 - do something else
- Part 3 - do the part 3 thing
- etc.

**Pull request closed without merging**